crates/aptos-rest-client/src/lib.rs

```diff
@@ -555,16 +555,21 @@ impl Client {
                     ))?;
                 }
                 WaitForTransactionResult::Pending(state) => {
+                    reached_mempool = true;
                     if expiration_timestamp_secs <= state.timestamp_usecs / 1_000_000 {
-                        return Err(anyhow!("Transaction expired, it is guaranteed it will not be committed on chain.").into());
+                        return Err(anyhow!("Transaction expired. It is guaranteed it will not be committed on chain.").into());
                     }
                     chain_timestamp_usecs = Some(state.timestamp_usecs);
                 }
                 WaitForTransactionResult::NotFound(error) => {
                     if let RestError::Api(aptos_error_response) = error {
                         if let Some(state) = aptos_error_response.state {
                             if expiration_timestamp_secs <= state.timestamp_usecs / 1_000_000 {
-                                return Err(anyhow!("Transaction expired, it is guaranteed it will not be committed on chain.").into());
+                                if reached_mempool {
+                                    return Err(anyhow!("Transaction expired. It is guaranteed it will not be committed on chain.").into());
```

Comment:
Could we still use a different error message here, so we can distinguish between NotFound and Pending?

Reply:
Spoke offline, we'll add a comment explaining what's going on here.